docs/DeveloperGuide.md

```diff
@@ -1,37 +1,293 @@
+Given below is an example usage scenario and how View Project behaves at each step
+
+**Step 1.** The user executes ```view cs2113``` to view the project titled cs2113. The command is retrieved and parsed by ```handleUserInput()``` in ```CommandHandler```, which calls ```view()``` in ```CommandHandler```
+
+**Step 2.** The ```view()``` method ensures that ```ProjectList``` has been correctly initialised and the command is valid, before calling ```view()``` in ```ProjectList```
+
+**Step 3.** ```view()``` in ```ProjectList``` calls ```findProjectIndex()``` in ```ProjectList```
+
+**Step 4.** ```findProjectIndex()``` loops through all ```Project``` objects in projectList until a ```Project``` with the same title as the one entered by the user is found. The ```Project```’s index is returned to ```CommandHandler```'s implementation of ```view()```
+
+**Step 5.** ```view()``` calls the ```printDetails()``` method of the ```Project``` at the given index
 
-## Design & implementation
+**Step 6.** ```printDetails()``` displays project title and deadline to the user. It iterates through the ```Todo```s in the ```todos``` attribute and prints each `Todo` by calling `Todo`’s `toString()` function
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![image info](./UmlDiagrams/ViewProject.jpg)
 
+#### Add a Deadline to a Project
+
+**Step 1.** When `CommandHandler` receives a user input starting with string “addprojdeadline”, it will create an `AddProjectDeadlineCommandParser` object and call its `parse()` function to parse the user input
+
+**Step 2.** The `AddProjectDeadlineCommandParser` object will parse the todo input. Upon parsing the user input, it will create a `AddProjectDeadlineCommand` and return it back to `CommandHandler`.
+
+**Step 3.** After `CommandHandler` obtains the `AddProjectDeadlineCommand` returned by `AddProjectDeadlineCommandParser`, it then calls `executeCommand()` method in its own class.
+
+**Step 4.** `executeCommand()` will call `execute()` method of `AddProjectDeadlineCommand`.
+
+**Step 5.** If the number of arguments is valid, call the projectList’s `addProjectDeadline()` method, with the arguments of title and deadline.
+
+**Step 6.** We call the `findProjectIndex()` method to find the given project that matches the title provided. If the value indicates that the project does not exist, return immediately and indicate no deadline was added.
+
+**Step 7.** If the index of the project was found, call that specific Project’s `setDeadline()` method and update the deadline.
+
+**Step 8.** The string deadline is used to build the class object `Deadline`
+
+**Step 9.** Inside the constructor it considers 2 types of inputs, a day of the week or a date format of yyyy-mm-dd. If it is a day of the week, it will properly detail next day that day of the week from the current day.
+
+![image info](./UmlDiagrams/AddProjectDeadline.jpg)
```

Comment:
The arrow heads are exceeding the activation bars. The text is abit hard to view..